ls1intum/Artemis
# `General:` Improve support for small screens for assessment and code editor

### Checklist

### Motivation and Context
Currently, both the assessment layout as well as the code editor top bar do not support smaller screens. This causes problems in Orion since Orion's integrated browser is rather small. The buttons get pushed out of the screen and a vertical scollbar appears.

### Description
This PR adds the flex-wrap property to all relevant containers in the assessment layout and code editor headbar. Wherever necessary it also changes the margins, from the current left-only to all four sides. This is required to ensure all buttons align correctly, regardless where they wrap into a new line.

I recommend to review the code with whitespace changes deactivated.

### Steps for Testing

1. View the assessment detail view and the code editor under all circumstances with different browser sizes.
2. The assessment layout circumstances include:
    - Assessment of all exercise types
    - Assessment in Orion (this can also be tested in a regular browser by overriding the userAgent to include "Orion/1.1.1")
    - An additional banner shown when the assessment due date has passed
    - An additional button in second correction rounds in exams
    - An additional button in text and modelling exercises that allow to convert the submission to an example submission
3. The code editor circumstances include:
    - The assessment of programming exercises
    - The participation in programming exercises as a student (both normal an in exams)
    - The creation of programming exercises as an instructor ("Edit in Editor" button)
4. All views should wrap nicely upon reducing the browser size. The code editor only wraps if you collaps the file browser or insstrctions tab
5. I might have missed an occurence or edge case, if you know any other variation of the mentioned components, please let me know

### Review Progress
- Code Review
- Manual Tests

### Test Coverage
unchanged

### Screenshots

#### Assessment with normal screen size
![grafik](https://user-images.githubusercontent.com/61357727/134024707-516ec49a-6126-4a13-bce5-2d725cb93721.png)
#### Assessment with small screen size, old
![grafik](https://user-images.githubusercontent.com/61357727/134024844-e03d4076-38ab-4e9a-912c-d3aa781e6aa2.png)
#### Assessment with small screen size, new
![grafik](https://user-images.githubusercontent.com/61357727/134024597-98eb821a-8458-4bbb-beef-4714982880ae.png)
